lib/origen_testers/smartest_based_tester/base/processors/extract_flow_vars.rb

```diff
@@ -56,6 +56,18 @@ def on_if_job(node)
           end
           alias_method :on_unless_job, :on_if_job
 
+          def on_whenever(node)
+            flow_var, *nodes = *node
+            [flow_var].flatten.each do |f|
+              if f.value.is_a?(String) && f.value.to_s.match?(/^\$/)
```

Comment:
Why gate the flag creation only if its a case sensitive (starts with $) String?
each of these are valid inputs to create the flow variable "MY_FLOW_VAR":
```ruby
:MY_FLOW_VAR
"MY_FLOW_VAR"
:"$MY_FLOW_VAR"
"$MY_FLOW_VAR"
```

Reply:
I'm not so much worried about a "native" use case where those vars shouldn't be added and to my knowledge they probably should be, and other than the outputs themselves changing this isn't a breaking change which is why I'm leaning away from an opt-**in** flag.

The scenario I had in mind that an opt-**out** flag would alleviate is that since these variables were not being added, users may have developed custom solutions to add them in or otherwise handle it and could cause them to need to update their software when pulling in this change. So this way they'd just need to supply a flag instead of tweaking their internal tooling. 

Granted this is all hypothetical and likely an over-abundance of caution, so I'm not saying it should be a blocker, just an idea.